### ISSUE TITLE
refactoring fetchBb

### ISSUE THREAD

**Issue by shreyas1599** [git-touch/git-touch#158]
Hi, @pd4d10  [`fetchBb`](https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/models/auth.dart#L431-L451) and [`fetchBbJson`](https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/models/auth.dart#L453-L464) are two different methods. I need to add other HTTP methods(PUT, DELETE, PATCH) like I did for [Gitee](https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/models/auth.dart#L323-L372). Is it alright if I refactor this into a single `fetchBb` method and update the references across all files? Also was there any specific reason you used two methods(fetchBb and fetchBbJson) for bitbucket unlike the others?

**Comment by pd4d10** [comment]
> refactor this into a single fetchBb method

OK. BTW, the `http` lib seems a bit verbose when using the HTTP method other than 'GET'.

Currently it is:

```dart
http.get()
http.post()
```

But for the common request utility function, what we want is more like:

```dart
http.request(method: MethodEnum.GET)
http.request(method: MethodEnum.POST)
```

```

**Comment by shreyas1599** [comment]
Oh do you mean using this function: 
```
http.Request(method: String, URL: Uri)
```
and then Something like this?

```
enum Methods {
    GET, POST,...
}
extension parsdUri on Methods {
   Uri getParsedUri(params) {

   }
}

http.Request(method: methodName, Methods.GET.getParsedUri(params))
```

But the above seems will not make the existing code simpler. 
Are you talking more along these lines: https://github.com/dart-lang/language/issues/158 ?
Seems like we can't create methods in enums. Only using extensions we can call methods

**Comment by pd4d10** [comment]
Yeah, I mean if the `http` lib does already support this, the code would be simpler. For example, we could get rid of the `switch case` as follows:

https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/models/auth.dart#L333-L366

We could just pass the `method` instead, no matter if it is a `string` or `enum`

**Comment by pd4d10** [comment]
I think your proposal is fine. Let's just do the refactor as you supposed.
